src/Web/Program.cs

```diff
@@ -23,22 +23,41 @@
 
 builder.Logging.AddConsole();
 
+var azdTemplate= builder.Configuration["AZD_TEMPLATE"];
+
+if (azdTemplate == "enable"){
+    var credential = new ChainedTokenCredential(new AzureDeveloperCliCredential(), new DefaultAzureCredential());
+    builder.Configuration.AddAzureKeyVault(new Uri(builder.Configuration["AZURE_KEY_VAULT_ENDPOINT"]), credential);
+    builder.Services.AddDbContext<CatalogContext>(c =>
+    {
+        var connectionString = builder.Configuration[builder.Configuration["AZURE_SQL_CATALOG_CONNECTION_STRING_KEY"]];
+        c.UseSqlServer(connectionString, sqlOptions => sqlOptions.EnableRetryOnFailure());
+    });
+    builder.Services.AddDbContext<AppIdentityDbContext>(options =>
+    {
+        var connectionString = builder.Configuration[builder.Configuration["AZURE_SQL_IDENTITY_CONNECTION_STRING_KEY"]];
+        options.UseSqlServer(connectionString, sqlOptions => sqlOptions.EnableRetryOnFailure());
+    });
+}
+else{
+    Microsoft.eShopWeb.Infrastructure.Dependencies.ConfigureServices(builder.Configuration, builder.Services);
+}
```

Comment:
I'm not sure I follow how this works.  Seems like the first conditional is for prod, but the second is for local.  But I'm not sure how it works.  Can you explain?

Reply:
About judging whether the environment is prod or local: we define a variable `azdTemplate`, and add a new parameter `AZD_TEMPLATE` in the [appSettings](https://github.com/dotnet-architecture/eShopOnWeb/pull/844/files#diff-7ef659fc9cf6968e718894d300490b14ea7a52091e7d4bcffae3a5029ac721d4R63) of the main.bicep file. If it is a prod environment, its value is `enable`. If it is a local environment, it will be empty.

Besides, the reason why a judgement is added here:
- For the [prod part](https://github.com/dotnet-architecture/eShopOnWeb/pull/844/files#diff-8e81326b7cf998379ece2f89d87036af374163ec93c6366e7b700f3d40160026R28-R42), since the `SQL ConnectionString` is saved in the `KeyVault`, we wrote a separate method for configuring the SQL Server for it.
- For the [local part](https://github.com/dotnet-architecture/eShopOnWeb/pull/844/files#diff-8e81326b7cf998379ece2f89d87036af374163ec93c6366e7b700f3d40160026R43-R45), we continued to use its original method.

Reply:
@jongio Do you still have doubts about this code snippet?

Reply:
Okay, so I think the name is throwing me off.  We should instead detect if running locally using the built in asp.net variablef or that.

Reply:
Sorry for confusion, we updated the variable name to `Environment`.

- For the prod part: The `Environment` value will be `Prod`.
- For the local part: We also added `Environment` in local appsettings.json too in order to detect if running locally. And it's value will be `Local`.